### PR TITLE
Amitaifrey/rd 617 improve building

### PR DESCRIPTION
**Pull request by amitaifrey**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -56,215 +49,73 @@ func tryCallParentPersistentPreRun(cmd *cobra.Command, args []string) error {
...
-	ecrCommand, err := buildRegistryCommand("ecr", ecr, services)
+	releaseCommand, err := buildRegistryCommand("release")
 	if err != nil {
-		finishWithError("failed creating ecr command", err)
+		finishWithError("failed creating release command", err)
 	}
 
+	// so we don't require everyone to install yq directly
+	// thankfully it's written in Go and with cobra!
+	yqCmd := yqcmd.New()
```
Comment (HackerQacker) [inline main.go:117]: Nice

File: config.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+func (this GoServiceConfig) GetBaseImage() string {
+	if this.BaseImage != "" {
+		return this.BaseImage
+	}
+
+	return viper.GetString("base_image")
+}
+
+func getConfig(registryName, path string) (*Config, error) {
+	f, err := os.Open(path)
```
Comment (HackerQacker) [inline config.go:67]: Why do we do all of this block instead of just calling ` viper.ReadInConfig()`?

File: main.go [hunk 3]
```diff
@@ -56,215 +49,73 @@ func tryCallParentPersistentPreRun(cmd *cobra.Command, args []string) error {
...
 	viper.SetConfigType("yaml")
 	viper.AddConfigPath(".")
 	viper.SetDefault("base_image", defaultBaseImage)
 	viper.SetEnvPrefix(configEnvPrefix)
 	viper.AutomaticEnv()
-
 }
 
 func main() {
-	err := viper.ReadInConfig()
```
Comment (HackerQacker) [inline main.go:243]: ^^

**Review by HackerQacker — COMMENTED** [review]
Also, when creating a new repository in the ECR Don't we need to add the relevant policy?
It was an issue with the previous ippon version, but since creating a new ECR repo is pretty rare we decided not to implement it (yet).
With this change we create a repo for every namespace right?

File: config.go (hunk 2 above)
Comment (amitaifrey) [reply config.go:67]: Because that way we can have multiple config files - useful for separating different builds for regular/llm

**Pull request merged**